### ISSUE TITLE
FIFOQueue '_1_processed_queue' is closed and has insufficient elements (requested 7, current size 0)

### ISSUE THREAD

**Issue by AirAI** [ethereon/caffe-tensorflow#124]
The tensorflow model generated by caffe-tensorflow works well i**n tensorflow v0.8.0**. It returned some errors when I transplanted it **to tensorlfow v1.2.0.** The following are my steps.

1. Firstly changed the interface for some function from tensorflow v0.8.0 to tensorflow v1.2.0 ,eg., changed tf.pack() to tf.stack()  in dataset.py
2. It returned the following error after step 1.(Attention: The error came from dataset.py, and I add some debug information in dataset.py, so the line number may be different with the code in github)

Caused by op u'processed_queue_DequeueMany', defined at:
  File "predict.py", line 67, in <module>
    main()
  File "predict.py", line 62, in main
    predict_and_display(network,img_paths,path_en,fnorm,fporn)
  File "predict.py", line 16, in predict_and_display
    categorys,confidences=network.predict(img_paths)
  File "/home/test/workspace/github/example/v0.2/example.py", line 43, in predict
    image_producer =  models.dataset.ImageProducer(image_paths=image_paths, data_spec=self.spec)
  **File "/home/test/workspace/github/example/v0.2/models/dataset.py", line 58, in __init__
    self.setup(batch_size=batch_size, num_concurrent=num_concurrent)
  File "/home/test/workspace/github/example/v0.2/models/dataset.py", line 108, in setup
    self.dequeue_op = processed_queue.dequeue_many(batch_size)**
  File "/home/test/anaconda2/lib/python2.7/site-packages/tensorflow/python/ops/data_flow_ops.py", line 457, in dequeue_many
    self._queue_ref, n=n, component_types=self._dtypes, name=name)
  File "/home/test/anaconda2/lib/python2.7/site-packages/tensorflow/python/ops/gen_data_flow_ops.py", line 1344, in _queue_dequeue_many_v2
    timeout_ms=timeout_ms, name=name)
  File "/home/test/anaconda2/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py", line 767, in apply_op
    op_def=op_def)
  File "/home/test/anaconda2/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 2534, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/home/test/anaconda2/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 1203, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

**OutOfRangeError (see above for traceback): FIFOQueue '_1_processed_queue' is closed and has insufficient elements (requested 7, current size 0)
	 [[Node: processed_queue_DequeueMany = QueueDequeueManyV2[component_types=[DT_INT32, DT_FLOAT], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](processed_queue, processed_queue_DequeueMany/n)]]**

**Comment by shrishtrived** [comment]
I am having the same problem. Can anyone tell how to fix it?

**Comment by shrishtrived** [comment]
example/imagenet/dataset.py
1.  tf.pack() -> tf.stack()
2.  tf.image.resize_images(img, new_shape[0], new_shape[2]) ->rf.image.resize_images(img,new_shape)
3. tf.reverse(img,[False,False,True]) -> tf.reverse(img,[2])

I was not doing the step 3 , it fixed my issue.

**Comment by AirAI** [comment]
This problem can be solved in issue #128 

The default version of tensorflow is v0.8.0. If you want to execture it in tensorflow v1.2.0,please do the following changes.

example/imagenet/dataset.py
tf.pack() -> tf.stack()
tf.image.resize_images(img, new_shape[0], new_shape[2]) -> rf.image.resize_images(img,new_shape)
tf.reverse(img,[False,False,True]) -> tf.reverse(img,[2])

kaffe/tensorfow/network.py
tf.concat(concat_dim=axis, values=inputs, name=name) -> tf.concat(axis=axis, values=inputs, name=name)

**Comment by GHmaryam** [comment]
I had a similar problem; I resolved it by changing the line:

`filename_queue = tf.train.string_input_producer([tfrecords_filename], num_epochs=num_epochs)
`

to 

`filename_queue = tf.train.string_input_producer([tfrecords_filename])
`

In the tensorflow document for tf.train.string_input_producer, it says:

> num_epochs: .... If not specified, string_input_producer can cycle through the strings in string_tensor an unlimited number of times.

That fixed my issue since I did not necessarily have this error in the first round; but very randomly on the subsequent epochs!

**Comment by nuclearwang** [comment]
I still have the same problem after changing the below line. It doesn't mean my training code cannot run. Actually, this problem appears after training about 100000 samples. Can anyone help me? Thanks in advance.

example/imagenet/dataset.py
tf.pack() -> tf.stack()
tf.image.resize_images(img, new_shape[0], new_shape[2]) -> rf.image.resize_images(img,new_shape)
tf.reverse(img,[False,False,True]) -> tf.reverse(img,[2])

kaffe/tensorfow/network.py
tf.concat(concat_dim=axis, values=inputs, name=name) -> tf.concat(axis=axis, values=inputs, name=name)
